CMakeLists.txt

```diff
@@ -40,11 +42,11 @@ if(CMAKE_Fortran_COMPILER_ID STREQUAL GNU)
   add_compile_options(-std=f2018)
 elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^Intel")
   if(WIN32)
-    add_compile_options(/warn:declarations,general,usage,interfaces,unused)
-    add_compile_options(/stand:f18)
+    add_compile_options("$<$<COMPILE_LANGUAGE:Fortran>:/warn:declarations,general,usage,interfaces,unused>")
+    add_compile_options("$<$<COMPILE_LANGUAGE:Fortran>:/stand:f18>")
   else()
-    add_compile_options(-warn declarations,general,usage,interfaces,unused)
-    add_compile_options(-stand f18)
+    add_compile_options("$<$<COMPILE_LANGUAGE:Fortran>:-warn declarations,general,usage,interfaces,unused>")
+    add_compile_options("$<$<COMPILE_LANGUAGE:Fortran>:-stand f18>")
```

Comment:
This doesn't work, the generator expression will produce a whole string `'-stand f18'` and the compiler will ignore the option.

Reply:
```
ifort: command line warning #10006: ignoring unknown option '-warn declarations,general,usage,interfaces,unused'
ifort: command line warning #10006: ignoring unknown option '-stand f18'
```

Reply:
I added these ones to avoid that these options are used by the Intel C/C++ compilers. However, these options are not recognized by the OneAPI ifort compiler. However, from the docs, it seems they should be.

Reply:
The generator expression will escape the space in the string, instead of two separate arguments `-stand f18` will be passed as `'-stand f18'`, *i.e.* as one argument, which is not recognized anymore.

Reply:
It would be nice if we could get rid of the `add_compile_options` completely and instead set those by the respective target and language specific CMake commands. Currently downstream CMake projects building stdlib on-demand (as subproject) will get all arguments from stdlib injected in their build, which makes it hard to use stdlib in any mixed language project at the moment.

Reply:
Would somehting like this:
```
    set(CMAKE_Fortran_FLAGS         "${CMAKE_Fortran_FLAGS} -stand f18")
```
be more appropriate?

Reply:
Dunno what the best way in CMake is, the current one is a bit suboptimal at best. For the moment let's fix the whitespace issue first and revisit the flags later.

Reply:
Sounds good.